pkg/crc/machine/bundle/metadata.go

```diff
@@ -40,25 +43,60 @@ type CrcBundleInfo struct {
 			Format string `json:"format"`
 		} `json:"diskImages"`
 	} `json:"storage"`
+	cachedPath string
 }
 
-func GetCrcBundleInfo(bundlePath string) (*CrcBundleInfo, string, error) {
-	var bundleInfo CrcBundleInfo
-	extractedPath, err := Extract(bundlePath, constants.MachineCacheDir)
+func getCachedBundlePath(bundleName string) string {
+	path := strings.TrimSuffix(bundleName, ".crcbundle")
+	return filepath.Join(constants.MachineCacheDir, path)
+}
+
+func (bundle *CrcBundleInfo) isCached() bool {
+	_, err := os.Stat(bundle.cachedPath)
+	return err == nil
+}
+
+func (bundle *CrcBundleInfo) readBundleInfo() error {
+	bundleInfoPath := bundle.resolvePath("crc-bundle-info.json")
+	f, err := ioutil.ReadFile(bundleInfoPath)
 	if err != nil {
-		return nil, extractedPath, fmt.Errorf("Error during extraction : %+v", err)
+		return fmt.Errorf("Error reading %s file : %+v", bundleInfoPath, err)
 	}
-	BundleInfoPath := filepath.Join(extractedPath, "crc-bundle-info.json")
-	f, err := ioutil.ReadFile(BundleInfoPath)
+
+	err = json.Unmarshal(f, bundle)
 	if err != nil {
-		return nil, extractedPath, fmt.Errorf("Error reading %s file : %+v", BundleInfoPath, err)
+		return fmt.Errorf("Error Unmarshal the data: %+v", err)
 	}
 
-	err = json.Unmarshal(f, &bundleInfo)
+	return nil
+}
+
+func GetCachedBundleInfo(bundleName string) (*CrcBundleInfo, error) {
+	var bundleInfo CrcBundleInfo
+	bundleInfo.cachedPath = getCachedBundlePath(bundleName)
+	if !bundleInfo.isCached() {
+		return nil, fmt.Errorf("Could not find cached bundle info")
+	}
+	err := bundleInfo.readBundleInfo()
 	if err != nil {
-		return nil, extractedPath, fmt.Errorf("Error Unmarshal the data: %+v", err)
+		return nil, err
 	}
-	return &bundleInfo, extractedPath, nil
+	return &bundleInfo, nil
+}
+
+/*
+func Extract(sourcepath string) (*CrcBundleInfo, error) {
+	err := extract.Uncompress(sourcepath, constants.MachineCacheDir)
+	if err != nil {
+		return nil, err
+	}
+
+	return GetCachedBundleInfo(filepath.Base(sourcepath))
+}
+*/
+
```

Comment:
@cfergeau does it still work in progress?

Reply:
Left-over during a rebase, thanks for spotting it, it's gone from the latest revision of the PR